### PR TITLE
#2994 Launch Page Unit Tests

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:*
https://github.com/bcgov/entity/issues/2994

*Description of changes:*

* Implements Unit testing for additional Views and components.

* Fixed some styling.

* Fixed some duplicate id's.

* Removed some unused imports.

*Checklist:*
I confirm that below checklist items are addressed in this pull request; (check the boxes applicable)
- [ ] No lint errors
- [ ] No test case failures and proper coverage for all modules/classes
- [ ] Updated the deployment configs for new environment variable(s)
- [ ] Updtaed the postman collection in entity repository (https://github.com/bcgov/entity/tree/master/api-e2e/postman) for e2e tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/tests/unit/views/HomeView.spec.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+      }
+    })
+
+    wrapper = mount(HomeViewDev, {
+      store,
+      localVue,
+      router,
+      vuetify,
+      mocks: {
+        $t: (mock) => mock
```
Comment (severinbeauvais) [inline auth-web/tests/unit/views/HomeView.spec.ts:67]: If I understand correctly, this means there's a function $t which, given a value, returns that value. Why is this needed?

File: auth-web/tests/unit/views/HomeView.spec.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+import Vuex from 'vuex'
+
+Vue.use(Vuetify)
+Vue.use(VueRouter)
+const router = new VueRouter()
+const vuetify = new Vuetify({})
+
+// Prevent the warning "[Vuetify] Unable to locate target [data-app]"
+document.body.setAttribute('data-app', 'true')
+jest.mock('../../../src/plugins/i18n', () => {})
```
Comment (severinbeauvais) [inline auth-web/tests/unit/views/HomeView.spec.ts:19]: Would `@` work here to replace "../../../src" ? Wondering if there's a specific requirement for this in the tests.

File: auth-web/tests/unit/views/HomeView.spec.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    expect(wrapper.find(BcscPanel).exists()).toBe(true)
+  })
+
+  it('renders the correct buttons when authenticated', () => {
+    const bannerBtns = wrapper.vm.$el.querySelectorAll('.cta-btn')
+    const nameRequestBtn = bannerBtns[0]
+    const namedCompBtn = bannerBtns[1]
+    const numberedCompBtn = bannerBtns[2]
+    const manageBusinessBtn = bannerBtns[3]
+    const loginBtn = bannerBtns[4]
```
Comment (severinbeauvais) [inline auth-web/tests/unit/views/HomeView.spec.ts:91]: Oh, I like this way of naming the buttons up front! Makes the 'expect' code much easier to read :)

File: auth-web/src/components/auth/BcscPanel.vue [hunk 4]
```diff
@@ -13,7 +13,7 @@
       <!-- Info Column -->
       <v-col cols="12" md="6" class="bcsc-info-col">
         <h2>It's Secure</h2>
-        <v-list-item class="list-item" v-for="(item, index) in secureBulletPoints" :key="index" >
+        <v-list-item class="list-item" v-for="(item, index) in secureBulletPoints" :key="`List-1-${index}`" >
```
[the remark this answers is not in the document]
Comment (cameron-freshworks) [reply auth-web/src/components/auth/BcscPanel.vue:16]: oh nice! This is good to know, thanks 👍

File: auth-web/tests/unit/components/BcscPanel.spec.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        vuetify,
+        propsData: {
+          ...propsData
+        }
+      })
+    }
+
+    wrapper = wrapperFactory({ userProfile: {} })
+
+    jest.resetModules()
```
[the remark this answers is not in the document]
Comment (cameron-freshworks) [reply auth-web/tests/unit/components/BcscPanel.spec.ts:37]: I can certainly echo that, sometimes what works in one test suite isn't always as expected in another. 
In this case, with the way auth-web has establish their store into modules, it seems to do the trick effectively. I will test this out tho!

File: auth-web/tests/unit/views/HomeView.spec.ts (hunk 1 above)
Comment (cameron-freshworks) [reply auth-web/tests/unit/views/HomeView.spec.ts:67]: It just defines (mocks i should say) `$t` as a global method, otherwise when it tries to render `$t` it has no reference and will return `"TypeError: _vm.$t is not a function"`.

File: auth-web/tests/unit/views/HomeView.spec.ts (hunk 2 above)
Comment (cameron-freshworks) [reply auth-web/tests/unit/views/HomeView.spec.ts:19]: Yes, good point! `@` would certainly work here. 

To be honest, i'm not sure 100% on this mock. I would have expected this mock to way for the mounted component to understand the global methods but it didn't work as how i expected. Let me do a little more research on this mock.

File: auth-web/tests/unit/views/HomeView.spec.ts (hunk 3 above)
Comment (cameron-freshworks) [reply auth-web/tests/unit/views/HomeView.spec.ts:91]: awesome, thanks!

File: auth-web/tests/unit/views/HomeView.spec.ts (hunk 1 above)
Comment (cameron-freshworks) [reply auth-web/tests/unit/views/HomeView.spec.ts:67]: > Great job Cam!

lol thanks! This message popped up about 7 times :) Really appreciate it heheh

File: auth-web/tests/unit/components/BcscPanel.spec.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (cameron-freshworks) [reply auth-web/tests/unit/components/BcscPanel.spec.ts:37]: refactored in next commit. implemented in `afterEach()` 👍

File: auth-web/tests/unit/views/HomeView.spec.ts (hunk 2 above)
Comment (cameron-freshworks) [reply auth-web/tests/unit/views/HomeView.spec.ts:19]: So, this wasn't required. I think? the intention was to give definition to such things such as the global methods ( ie $t ), but it didn't quite mock it as expected. Which is why i opted for the approach from above. Removed this in next commit!
